src/librustc/mir/transform.rs

```diff
@@ -70,16 +72,32 @@ impl<'a, 'tcx> MirSource {
 
 /// Various information about pass.
 pub trait Pass {
-    // fn name() for printouts of various sorts?
     // fn should_run(Session) to check if pass should run?
     fn dep_node(&self, def_id: DefId) -> DepNode<DefId> {
         DepNode::MirPass(def_id)
     }
+    fn name(&self) -> &str;
```

Comment:
A default implementation of this could be `::std::intrinsics::type_name::<Self>()`. With that as a default, I doubt many passes would like to manually implement this method.